scripts/test-e2e-local.js

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ */
+
+'use strict';
+
+/*
+ * This script is a re-interpretation of the old test-manual.e2e.sh script.
+ * the idea is to provide a better DX for the manual testing.
+ * It's using Javascript over Bash for consistency with the rest of the recent scripts
+ * and to make it more accessible for other devs to play around with.
+ */
+
+const {exec, exit, pushd, popd, pwd, cd} = require('shelljs');
+const yargs = require('yargs');
+const fs = require('fs');
+const path = require('path');
+const os = require('os');
+
+const {
+  launchAndroidEmulator,
+  isPackagerRunning,
+  launchPackagesInSeparateWindow,
+} = require('./testing-utils');
+
+const {generateAndroidArtifacts} = require('./release-utils');
+
+// setting this just 'cause this https://github.com/facebook/react-native/commit/0a3ca80af401654896cbc73e235711eef0f9b3c5
+// breaks a bunch of stuff. This should be enough.
+// TODO: follow up with Nicola in understanding the impact of the commit, and if we can remove this
+// also, remove the folder
+process.env.TMP_PUBLISH_DIR = fs.mkdtempSync(
+  path.join(os.tmpdir(), 'rn-publish-'),
+);
```

Comment:
cc @vincenzovitale

Reply:
@kelset 
#34746 will prevent the setting of 
process.env.TMP_PUBLISH_DIR